prime-router/docs/allowlist-gitleaks-false-positives.md

```diff
@@ -0,0 +1,158 @@
+* Check out a new branch off of the "`master`" branch (let's call this one "`gitleaks-culprit`").
+* In "`gitleaks-culprit`", modify the `gitleaks-config.toml` file to have the correct suppressions.
+* Push up the "`gitleaks-culprit`" branch and open a Pull Request, include at least one member of the DevOps team (e.g. @ronaldheft-usds or @td-usds). The Pull Request _must_ contain an explanation of:
+    * What you are suppressing and why it's ok: the specific patterns/values/commits; example:
+        * "Gitleaks correctly flagged the Private Key pattern, but this is a test key so it's not sensitive."
+    * How you ensured the smallest possible scope for the suppression, you may be asked to narrow the scope; example:
+        * "This commit does not contain any other code except for the addition of test keys, there is nothing else in this commit to be flagged. The suppression disables this particular rule for just this commit. Other rules still apply."
+    * Why the suppression will not cause False Negatives; example:
+        * "Suppressing the pattern "`it\.key\.contains\(`" in the "Generic Credential" rule set is fine because this value is not a credential, it is a key lookup in an iterator."
+* Get approval for the PR and merge the PR into "`master`" on approval by at least one member of the DevOps team, when the nightly run kicks off, it will now have your suppression(s) applied and not raise a false positive on the changes in your branch.
+* To bring the suppression into your own branch, either `git merge master` into your branch or `git cherrypick` the commit that contains the suppression on to your branch.
+
+# How to Suppress
+
+It is crucial to scope the suppression as small as possible. Suppressions that are too broad may inadvertently cause False Negatives (cases where a sensitive value should have been flagged, but isn't because it matches an exclusion). _While false positives can be annoying, false negatives can be really bad_!
+
+Gitleaks allows for the definition of rules by RegEx and allows for suppressions of that specific rule based on one of the following mechanisms:
+
+* By commit: do not apply this rule on the commit with hash "`H`"
+* By pattern: if a line matches your rule, but _also_ matches this other pattern, then do not flag this as a violation of the rule
+* By file or path: do not apply this rule on these files or files under this path
+
+_As a courtesy to your fellow developers_, when putting suppressions in place, if the suppression involves adding to a list, always terminate each added line with a comma (allowed by TOML) so that each addition (or removal) now or in the future shows up as a single line modification in the source code diffs.
+
+## By Commit
+
+As long as you "commit small, commit often", this is _typically_ the best way to put a suppression in place. However, this does require that your code is already commited (which may not always be the case).
```

Comment:
The commit method I'm not sure is the best. Won't merge commits be flagged, therefore you would need to know the commit in your branch and the future upcoming, unknown merge commit id?

This will be flagged in two areas:
- The merge to master/main
- The merge to production

Reply:
Correct... I will add guidance to not use commit-based exclusions! Good point!

Reply:
fixed through 064bb73